From 5f84f6265b2b24af025759d6f01874b10ccf5ade Mon Sep 17 00:00:00 2001
From: chenj <chenj@lemote.com>
Date: Thu, 1 Dec 2016 09:51:35 +0800
Subject: [PATCH 218/457] AOSCOS: snd/hda/conexant: add support for raw verbs

[Mingcong Bai: Resolved a minor merge conflict in
sound/pci/hda/patch_conexant.c.]

Signed-off-by: Huacai Chen <chenhc@lemote.com>
Signed-off-by: Hongliang Tao <taohl@lemote.com>
Signed-off-by: Hua Yan <yanh@lemote.com>
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>

Signed-off-by: chenj <chenj@lemote.com>
Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
Signed-off-by: ilikara <3435193369@qq.com>
---
 sound/pci/hda/patch_conexant.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/sound/pci/hda/patch_conexant.c b/sound/pci/hda/patch_conexant.c
index 0cec1344344a..c7801a0a2e43 100644
--- a/sound/pci/hda/patch_conexant.c
+++ b/sound/pci/hda/patch_conexant.c
@@ -43,6 +43,9 @@ struct conexant_spec {
 	unsigned int gpio_mute_led_mask;
 	unsigned int gpio_mic_led_mask;
 	bool is_cx11880_sn6140;
+
+	const unsigned int *raw_init_verbs[5];
+	unsigned int num_raw_init_verbs;
 };
 
 
@@ -184,7 +187,13 @@ static void cx_fixup_headset_recog(struct hda_codec *codec)
 
 static int cx_auto_init(struct hda_codec *codec)
 {
+	int i, j;
 	struct conexant_spec *spec = codec->spec;
+
+	for (i = 0; i < spec->num_raw_init_verbs; i++)
+		for (j = 0; spec->raw_init_verbs[i][j] != -1; j++)
+			snd_hda_codec_exec_verb(codec, spec->raw_init_verbs[i][j], NULL);
+
 	snd_hda_gen_init(codec);
 	if (!spec->dynamic_eapd)
 		cx_auto_turn_eapd(codec, spec->num_eapds, spec->eapds, true);
-- 
2.55.0

